github/github.go

```diff
@@ -2,55 +2,190 @@ package github
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
 	"github.com/dgrijalva/jwt-go"
 	"github.com/kintohub/utils-go/klog"
 	"github.com/valyala/fasthttp"
-	"strings"
-	"time"
 )
 
 type GithubInterface interface {
-	CallGithub(endpoint, verb, params, authToken string) ([]byte, error)
-	GenerateInstallationToken(installationId string) (string, error)
-	GenBearer(jwtToken string) string
+	GetListRepos(page int32, installationId, userAccessToken string) (*GithubRepositories, error)
+	GetUserInformation(userAccessToken string) (*GithubUserInfo, error)
+	CreateGithubAppToken(installationId string) (string, error)
+	GetUserAccessToken(code string) (string, error)
 }
 
 type github struct {
-	baseUrl       string
-	acceptHeader  string
-	appID         string
-	appPrivateKey []byte
+	appId           string
+	appClientId     string
+	appClientSecret string
+	appPrivateKey   []byte
 }
 
-func New(baseUrl, acceptHeader, appID string, appPrivateKey []byte) GithubInterface {
+var (
+	BASE_URL                 = "https://api.github.com"
+	BASE_URL_PLAIN           = "https://github.com"
```

Comment:
comment: for simplicity maybe we can call them `BASE_API_URL` and `BASE_URL` ?

Reply:
Its really weird man, coz both of them are used in API calls, I think eventually `BASE_URL_PLAIN` will be removed but currently, there is one GitHub endpoint that still uses it

Reply:
That being said ill rename them, i like ur names better